packages/@aws-cdk/aws-ecs/test/ec2/ec2-service.test.ts

```diff
@@ -3280,7 +3282,106 @@ describe('ec2 service', () => {
   });
 
   describe('When import an EC2 Service', () => {
-    test('with serviceArn', () => {
+    test('fromEc2ServiceArn old format', () => {
+      // GIVEN
+      const stack = new cdk.Stack();
+
+      // WHEN
+      const service = ecs.Ec2Service.fromEc2ServiceArn(stack, 'EcsService', 'arn:aws:ecs:us-west-2:123456789012:service/my-http-service');
+
+      // THEN
+      expect(service.serviceArn).toEqual('arn:aws:ecs:us-west-2:123456789012:service/my-http-service');
+      expect(service.serviceName).toEqual('my-http-service');
+    });
+
+    test('fromEc2ServiceArn new format', () => {
+      // GIVEN
+      const stack = new cdk.Stack();
+
+      // WHEN
+      const service = ecs.Ec2Service.fromEc2ServiceArn(stack, 'EcsService', 'arn:aws:ecs:us-west-2:123456789012:service/my-cluster-name/my-http-service');
+
+      // THEN
+      expect(service.serviceArn).toEqual('arn:aws:ecs:us-west-2:123456789012:service/my-cluster-name/my-http-service');
+      expect(service.serviceName).toEqual('my-http-service');
+    });
+
+    describe('fromEc2ServiceArn tokenized ARN', () => {
+      testLegacyBehavior(
+        'when @aws-cdk/aws-ecs:arnFormatIncludesClusterName is disabled, use old ARN format',
+        cdk.App,
+        (customApp) => {
+          // GIVEN
+          const stack = new cdk.Stack(customApp);
+
+          // WHEN
+          const service = ecs.Ec2Service.fromEc2ServiceArn(stack, 'EcsService', new cdk.CfnParameter(stack, 'ARN').valueAsString);
+
+          // THEN
+          expect(stack.resolve(service.serviceArn)).toEqual({ Ref: 'ARN' });
+          expect(stack.resolve(service.serviceName)).toEqual({
+            'Fn::Select': [
+              1,
+              {
+                'Fn::Split': [
+                  '/',
+                  {
+                    'Fn::Select': [
+                      5,
+                      {
+                        'Fn::Split': [
+                          ':',
+                          { Ref: 'ARN' },
+                        ],
+                      },
+                    ],
```

Comment:
Oof, this is very hard to determine if this is correct (although I am pretty sure it is). And unfortunately, I don't think there is another way to implement this to make the resulting template more readable.

Reply:
Should I add integration tests that show the result using CloudFormation Outputs? I tested the changes using this code:

```ts
const cluster = new ecs.Cluster(this, 'Cluster', {
  vpc,
});

const param = new CfnParameter(this, 'Param', {
  default: 'arn:aws:ecs:us-west-2:123456789012:service/my-http-service',
})

const fromFargateServiceArn = ecs.FargateService.fromFargateServiceArn(this, 'fromFargateServiceArn', param.valueAsString);
new CfnOutput(this, 'fromFargateServiceArnOutput', {
  value: fromFargateServiceArn.serviceName,
});

const fromEc2ServiceArn = ecs.Ec2Service.fromEc2ServiceArn(this, 'fromEc2ServiceArn', param.valueAsString);
new CfnOutput(this, 'fromEc2ServiceArnOutPut', {
  value: fromEc2ServiceArn.serviceName,
});

const fromEc2ServiceAttributes = ecs.Ec2Service.fromEc2ServiceAttributes(this, 'fromEc2ServiceAttributes', {
  cluster,
  serviceArn: param.valueAsString,
});
new CfnOutput(this, 'fromEc2ServiceAttributesOutput', {
  value: fromEc2ServiceAttributes.serviceName,
});

const fromFargateServiceAttributes = ecs.FargateService.fromFargateServiceAttributes(this, 'fromFargateServiceAttributes', {
  cluster,
  serviceArn: param.valueAsString,
});
new CfnOutput(this, 'fromFargateServiceAttributesOutput', {
  value: fromFargateServiceAttributes.serviceName,
});
```

New ARN Format and feature flag enabled (`CfnParameter`: `arn:aws:ecs:us-west-2:123456789012:service/cluster-name/my-http-service`) - Output:

```
MyStack.fromEc2ServiceArnOutPut = my-http-service
MyStack.fromEc2ServiceAttributesOutput = my-http-service
MyStack.fromFargateServiceArnOutput = my-http-service
MyStack.fromFargateServiceAttributesOutput = my-http-service
```

Old ARN Format and feature flag disabled (`CfnParameter`: `arn:aws:ecs:us-west-2:123456789012:service/my-http-service`) - Output:
```
MyStack.fromEc2ServiceArnOutPut = my-http-service
MyStack.fromEc2ServiceAttributesOutput = my-http-service
MyStack.fromFargateServiceArnOutput = my-http-service
MyStack.fromFargateServiceAttributesOutput = my-http-service
```